app/assets/javascripts/media_player_wrapper/avalon_player_new.es6

```diff
@@ -223,11 +223,70 @@ class MEJSPlayer {
         paused: false
       };
 
-      // Go to next section
-      this.getNewStreamAjax(
-        sectionId,
-        `/media_objects/${mediaObjectId}/section/${sectionId}`
-      );
+      // Get the new stream info
+      $('.media-show-page').removeClass('ready-to-play');
+        $.ajax({
+          url: `/media_objects/${mediaObjectId}/section/${sectionId}/stream`,
+          dataType: 'json',
+          data: {
+            content: sectionId
+          }
+        })
+          .done(response => {
+            this.setContextVars(response);
+            // Switch media in the current player instance
+            this.switchMedia(mediaElement, originalNode, instance, response);
+            this.updateShareLinks();
+          })
+          .fail(error => {
+            console.log('error', error);
+          });
+    }
+  }
+
+  /**
+   * Swap the media source and track in the current player instance with values from
+   * the new stream info
+   * @param media {Object} - MediaElement wrapper
+   * @param node {Object} - HTML node
+   * @param instance {Object} - player instance
+   */
+  switchMedia(media, node, instance) {
+    let markup = '';
+
+    node.innerHTML = '';
+    this.currentStreamInfo.stream_hls.map(source => {
+      markup += `<source src="${
+        source.url
+      }" type="application/x-mpegURL" data-quality="${source.quality}"/>`;
```

Comment:
I think it is okay to hardcode the type here but ideally I think we would want to get it from the stream info returned from the server (if it does return that information).

Reply:
I will look through `currentStreamInfo` and see whether it has the type coming in with the stream.

Reply:
@cjcolvar `currentStreamInfo` has a `mimetype` in its data for each source. But they have `null` as the value.